src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -430,6 +478,62 @@
                              jruby-service)
         (environment-not-found environment)))))
 
+(defn exception-matches?
+  [^Exception e pattern]
+  (->> e
+       .getMessage
+       (re-find pattern)
+       boolean))
+
+(schema/defn ^:always-validate
+  task-details :- TaskDetails
```

Comment:
maybe a short docstring would be appropriate here?

Reply:
yes, the naming in this file is confusing, having both describe-task which returns TaskDetails and task-details which returns TaskDetails, I'd have no idea what does what from just looking at that